### DISCUSSION TITLE
TensorBoard?

### DISCUSSION THREAD

**Discussion by desaibhargav** [buildingamind/NewbornEmbodiedTuringTest#144] [Ideas]
I just wanted to get people's thoughts on whether TensorBoard integration would be a good thing to have (or even make sense). [TensorBoard](https://www.tensorflow.org/tensorboard) is basically a real-time dashboard for (pretty much) all things related to model training and I've personally found it useful for debugging. 

An example of what this looks like: https://huggingface.co/bigscience/bloom/tensorboard 

A central dashboard for monitoring and logging experiments feels like a good to have for the number of experiments we run. It IS a little finicky to set up, but I can take initiative to get it working for us if everyone is onboard with the idea. 

Some use-cases (other than the obvious ones) that come to mind are: 

- Adding short [videos](https://pytorch.org/docs/stable/tensorboard.html#torch.utils.tensorboard.writer.SummaryWriter.add_video) of agents while the training is in progress. It may be useful debugging such inspecting the videos with the reward signals side-by-side.
- Since it's all real-time, it may be useful to pull plugs on experiments if necessary and save time. Plus, a lot of things can be logged and monitored (such as @Zach-Attach's very cool density map of the area covered by the agents). It can log whatever we want, audio, video, text or scalars. 

What do we think?

**Comment by desaibhargav** [comment]
The idea of a TV monitor with live images / videos / high-level conceptual metrics from various experiments is growing on me 😵‍💫
